test/system/src/main/java/io/pravega/test/system/framework/services/docker/BookkeeperDockerService.java

```diff
@@ -0,0 +1,103 @@
+import com.spotify.docker.client.messages.swarm.ContainerSpec;
+import com.spotify.docker.client.messages.swarm.EndpointSpec;
+import com.spotify.docker.client.messages.swarm.NetworkAttachmentConfig;
+import com.spotify.docker.client.messages.swarm.PortConfig;
+import com.spotify.docker.client.messages.swarm.ResourceRequirements;
+import com.spotify.docker.client.messages.swarm.Resources;
+import com.spotify.docker.client.messages.swarm.RestartPolicy;
+import com.spotify.docker.client.messages.swarm.ServiceMode;
+import com.spotify.docker.client.messages.swarm.ServiceSpec;
+import com.spotify.docker.client.messages.swarm.TaskSpec;
+import java.net.URI;
+import lombok.extern.slf4j.Slf4j;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import static io.pravega.test.system.framework.Utils.DOCKER_NETWORK;
+
+@Slf4j
+public class BookkeeperDockerService extends DockerBasedService {
+
+    private static final int BK_PORT = 3181;
+    private final long instances = 3;
+    private final double cpu = 0.1;
+    private final double mem = 1024.0;
+    private final URI zkUri;
+
+    public BookkeeperDockerService(String serviceName, final URI zkUri) {
+        super(serviceName);
+        this.zkUri = zkUri;
+    }
+
+    public void stop() {
+        super.stop();
+    }
+
+    @Override
+    public void clean() {
+    }
+
+    @Override
+    public void start(final boolean wait) {
+        start(wait, setServiceSpec());
+    }
+
+    private ServiceSpec setServiceSpec() {
+
+        Map<String, String> labels = new HashMap<>();
+        labels.put("com.docker.swarm.service.name", serviceName);
+
+        Mount mount1 = Mount.builder().type("volume").source("index-volume").target("/bk/index")
+                .build();
+        Mount mount2 = Mount.builder().type("volume").source("logs-volume")
+                .target("/opt/dl_all/distributedlog-service/logs/")
+                .build();
+        //TODO: add journal and ledger mounted volumes
+        String zk = zkUri.getHost() + ":" + ZKSERVICE_ZKPORT;
+        List<String> stringList = new ArrayList<>();
+        String env1 = "ZK_URL=" + zk;
+        String env2 = "ZK=" + zk;
+        String env3 = "bookiePort=" + String.valueOf(BK_PORT);
+        String env4 = "DLOG_EXTRA_OPTS=-Xms512m";
+        stringList.add(env1);
+        stringList.add(env2);
+        stringList.add(env3);
+        stringList.add(env4);
+        final TaskSpec taskSpec = TaskSpec
+                .builder().restartPolicy(RestartPolicy.builder().maxAttempts(0).condition("none").build())
+                .containerSpec(ContainerSpec.builder()
+                        .hostname(serviceName)
+                        .labels(labels)
+                        .image(IMAGE_PATH + "/nautilus/bookkeeper:" + PRAVEGA_VERSION)
+                        .healthcheck(ContainerConfig.Healthcheck.create(null, 1000000000L, 1000000000L, 3))
```

Comment:
What are these large values in this health check? Is this right?

Reply:
Yes, it is . The create()  accepts parameters in nano seconds` create(cmd, interval, Timeout, retries).`

Reply:
If I'm reading this right, the interval and timeout values are of 1s, and at least the timeout seems short to me.

For clarity, consider doing something like:

```
Duration.ofSeconds(10).toNanos()
```

To represent a duration of 10 seconds in nanoseconds.

Reply:
done.